.travis.yml

```diff
@@ -1,17 +1,37 @@
 sudo: required
 dist: trusty
 
+services:
+  - docker
+
 language: java
 jdk:
   - openjdk8
   - oraclejdk8
 
 git:
   submodules: false
-before_install: scripts/travis-initialize-submodules
+before_install:
+  - scripts/travis-initialize-submodules
+  - scripts/run-xvfb
 
 install: ./gradlew setupDecompWorkspace
-script: ./gradlew build
+script: DISPLAY=:99 ./gradlew build runMcTester --stacktrace
+after_failure: scripts/upload
 
 notifications:
   email: false
+
+before_cache:
+  - rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
+  - rm -fr $HOME/.gradle/caches/*/plugin-resolution/
+cache:
+  directories:
+    - $HOME/.gradle/caches/
+    - $HOME/.gradle/wrapper/
+    - $TRAVIS_BUILD_DIR/.gradle/
+    - $TRAVIS_BUILD_DIR/build/
```

Comment:
Why do you cache the build files? This directory should be empty on each Travis build to ensure a clean re-build.

Additionally, files in there will definitely change with every run, and re-uploading is slow. With a bit of luck, builds will be as slow as without the cache. (Which is the main reason why we haven't been using the cache yet...)